server/src/main/java/org/candlepin/dto/manifest/v1/DistributorVersionDTO.java

```diff
@@ -0,0 +1,340 @@
+/**
+ * Copyright (c) 2009 - 2018 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto.manifest.v1;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.candlepin.dto.TimestampedCandlepinDTO;
+import org.candlepin.util.SetView;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * A DTO representation of the DistributorVersion entity as used by the manifest import/export framework.
+ */
+@XmlRootElement(name = "distributorversion")
+@XmlAccessorType(XmlAccessType.PROPERTY)
+public class DistributorVersionDTO extends TimestampedCandlepinDTO<DistributorVersionDTO> {
+    public static final long serialVersionUID = 1L;
+
+    /**
+     * Internal DTO object for DistributorVersionCapability
+     */
+    @XmlRootElement(name = "distributorversioncapability")
+    @XmlAccessorType(XmlAccessType.PROPERTY)
+    public static class DistributorVersionCapabilityDTO {
+        private String id;
+        private final String name;
+
+        @JsonCreator
+        public DistributorVersionCapabilityDTO(
+            @JsonProperty("id") String id,
+            @JsonProperty("name") String name) {
+            if (name == null || name.isEmpty()) {
+                throw new IllegalArgumentException(
+                    "The distributor version capability name is null or empty.");
+            }
+
+            this.id = id;
+            this.name = name;
+        }
+
+        public String getId() {
+            return this.id;
+        }
+
+        public void setId(String id) {
+            this.id = id;
+        }
+
+        public String getName() {
```

Comment:
This isn't a deal breaker, but there's no mutator for name. The only thing that must be immutable on these type of inner-DTOs are object references. If there's a reason to not have it, that's fine; but otherwise we should try to adhere to the DTO design requirements as much as possible.